lib/src/authentication/sign_in/sign_in_view.dart

```diff
@@ -0,0 +1,204 @@
+                        icon: Icon(passwordVisible
+                            ? Icons.visibility
+                            : Icons.visibility_off),
+                        onPressed: () {
+                          setState(
+                            () {
+                              passwordVisible = !passwordVisible;
+                            },
+                          );
+                        },
+                      ),
+                      alignLabelWithHint: false,
+                    ),
+                  ),
+                ),
+                SizedBox(
+                  width: double.infinity,
+                  child: Padding(
+                    padding: const EdgeInsets.only(
+                        right: 16.0, top: 8.0, bottom: 8.0),
+                    child: InkWell(
+                      key: const Key("sign_in_view_forgot_password"),
+                      onTap: viewModel.forgotPassword,
+                      child: Text(
+                        l10n.sign_in_view_forgot_password,
+                        textAlign: TextAlign.right,
+                        style: const TextStyle(fontSize: 16.0),
+                      ),
+                    ),
+                  ),
+                ),
+                SizedBox(
+                  width: double.infinity,
+                  child: Padding(
+                    padding: const EdgeInsets.all(12.0),
+                    child: ElevatedButton(
+                      key: const Key("sign_in_view_sign_in_button"),
+                      onPressed: viewModel.signIn,
+                      child: Text(
+                        l10n.sign_in_view_sign_in,
+                        style: const TextStyle(fontSize: 20.0),
+                      ),
+                    ),
+                  ),
+                ),
+                Padding(
+                  padding: const EdgeInsets.only(top: 28.0, bottom: 28.0),
+                  child: Row(
+                    children: [
+                      textDivider,
+                      Text(
+                        l10n.sign_in_view_or_separator,
+                        style: const TextStyle(fontSize: 20.0),
+                      ),
+                      textDivider
+                    ],
+                  ),
+                ),
+                Row(
+                  children: [
+                    Expanded(
+                      child: ElevatedButton(
+                        key: const Key("sign_in_view_google_button"),
+                        onPressed: viewModel.signInGoogle,
+                        style: ElevatedButton.styleFrom(
+                          shape: const CircleBorder(),
+                          side: const BorderSide(),
+                        ),
+                        child: SvgPicture.asset(
+                          "assets/images/logos/google_web_light_rd.svg",
+                          height: 56.0,
+                        ),
+                      ),
```

Comment:
Should be a separate widget to facilitate testing and reusability

Reply:
Updated